doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+
+    :param table allowed_values: allowed values of array items
+    :param table annotations: annotations (see :ref:`config_utils_schema_annotation`)
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_array`
+
+..  _config-utils-schema_object:
+
+schema_object
+~~~~~~~~~~~~~
+
+..  class:: schema_object
+
+    ..  _config-schema_object-apply_default:
+
+    ..  method:: apply_default(data)
+
+        Apply default values to scalar nodes. The functions takes the ``default``
+        built-in annotation values of the scalar nodes and applies them based
+        on the ``apply_default_if`` annotation. If there is no ``apply_default_if``
+        annotation on a node, the default value is also applied.
+
+        .. note::
+
+            The method works for static defaults. To define a dynamic default value,
+            use the :ref:`map() <config-schema_object-map>` method.
+
+        :param any data: configuration data
+
+        :return: configuration data with applied schema defaults
+
+        **See also:** :ref:`default <config-schema_node_annotation-default>`, :ref:`apply_default_if <config-schema_node_annotation-apply_default_if>`
+
+    ..  _config-schema_object-filter:
+
+    ..  method:: filter(data, f)
+
+        Filter data based on the schema annotations. The methods returns an iterator
+        by configuration nodes for which the given filter function ``f`` returns ``true``.
+
+        The filter function ``f`` receives the following table as the argument:
+
+        .. code-block:: lua
+
+            w = {
+                path = <array-like table>,
+                schema = <schema node>,
+                data = <data at the given path>,
+            }
+
+        The filter function returns a boolean value that is interpreted as "accepted" or "not accepted".
+
+        **Example:**
+
+        Calling a function on all schema nodes that have the ``my_annotation``
+        annotation defined:
+
+        .. code-block:: lua
+
+            s:filter(function(w)
+                return w.schema.my_annotation ~= nil
+            end):each(function(w)
+                do_something(w.data)
+            end)
+
+        :param any data: configuration data
+        :param function f: filter function
+
+        :return: a luafun iterator
+
+    ..  _config-schema_object-get:
+
+    ..  method:: get(data, path)
+
+        Get nested configuration values at the given path. The path can be
+        either a dot-separated string (``http.scheme``) or an array-like table (``{ 'http', 'scheme'}``).
+
+        **Example:**
+
+        ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_record_hierarchy/http_api.lua
+            :language: lua
+            :start-at: local scheme
+            :end-before: local host
+            :dedent:
+
+        :param any data: configuration data
+        :param any path: path to the target node
```

Comment:
From where the `path`'s type information comes? It is either a string in the dot-notation or an array-like table. Not something arbitrary.

Reply:
Fixed